src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,82 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
+    private static final HashMap<Class<?>, Class<?>> implementationMap = new HashMap<>();
+
+    static {
+        implementationMap.put(FileReaderService.class, FileReaderServiceImpl.class);
+        implementationMap.put(ProductParser.class, ProductParserImpl.class);
+        implementationMap.put(ProductService.class, ProductServiceImpl.class);
+    }
+
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        checkIfClassIsComponent(clazz);
+        Object clazzImplementationInstance = createNewInstance(clazz);
+        injectFields(clazzImplementationInstance);
+        return clazzImplementationInstance;
+    }
+
+    private void checkIfClassIsComponent(Class<?> clazz) {
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Class " + clazz.getName()
+                                                + " is missing @Component annotation.");
+        }
+    }
+
+    private void injectFields(Object instance) {
+        Field[] declaredFields = instance.getClass().getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                try {
+                    field.setAccessible(true);
+                    field.set(instance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Can't initialize field value. "
+                        + "Class " + instance.getClass().getName()
+                        + ", field " + field.getName());
```

Comment:
Don't forget to add an exception you're catching in catch block to the RuntimeException object you throw
```suggestion
                    throw new RuntimeException("Can't initialize field value. "
                        + "Class " + instance.getClass().getName()
                        + ", field " + field.getName(), e);
```